### ISSUE TITLE
Unable to link Docker Hub organization to Bitbucket repositories via service account

### ISSUE THREAD

**Issue by rcdailey** [docker/hub-feedback#1833]
I am following the directions to [link a service account](https://docs.docker.com/docker-hub/builds/#service-users-for-team-autobuilds) to bitbucket. I have created a service account, added it to the Bitbucket Team (as a group). And added that group to a few repositories, as Admin, for integration with Docker Hub Repositories. I was able to successfully link Docker Hub to Bitbucket via this service account on the Docker Hub Organization Settings page.

Next, I went to one of the existing repositories I have on Docker Hub. I clicked the "Builds" tab then selected "Link to Bitbucket (Connected)". After that, I am asked to pick a service account:

![image](https://user-images.githubusercontent.com/1768054/59942161-902b4880-9424-11e9-8b94-d1d6206374fd.png)

I selected the service account name in the "Select organization" drop down. After selecting, nothing happens for a few minutes. So I refresh the page and I see this:

![image](https://user-images.githubusercontent.com/1768054/59942210-a933f980-9424-11e9-8137-e46ea55fbc73.png)

I get the following error:

> Error trying to get GitHub user: 401 Client Error: Unauthorized for url: https://api.github.com/user

Why is it trying to reach out to GitHub? This is a Bitbucket integration. I did not connect the Docker Hub Organization to GitHub, so it shouldn't be used. I am not able to connect to my Bitbucket repositories because of this error. What am I doing wrong, or is this a bug?

**Comment by cowsrule** [comment]
Thanks for the report, we will look into it. Sounds like this is an issue on our end.

**Comment by philgebhardt** [comment]
We're seeing something similar for our Bitbucket service account. Unless the service account is added to the `Administrators` User group of the targeted organization, Docker Hubs dropdown for choosing the Bitbucket organization is missing what I'd expect.

*With Service account set as `ADMIN` for a specific repository*
DockerHub is unable to show the organization needed to link the repository for autobuilds

*With Service account added to the Administrators user group (`ADMIN` for all repositories)*
DockerHub is able to show the intended organization and all associated repositories

**Comment by d0meo** [comment]
I've tried to add a service account to an Administrators user group I created for both the repository and the workspace as below:
![image](https://user-images.githubusercontent.com/41641025/83356288-6a762580-a38f-11ea-9a52-c8d627e22c3b.png)
![image](https://user-images.githubusercontent.com/41641025/83356297-77931480-a38f-11ea-9125-78007a3f5525.png)
I've made both the repository and workspace public, but I still get no repositories to select from:
![image](https://user-images.githubusercontent.com/41641025/83356391-128bee80-a390-11ea-82a4-fae825bd97e8.png)

**Comment by mohammedchairi** [comment]
Hello team, I have the same issue.
Have you any idea what I have to do ?

**Comment by mkocus** [comment]
Any news regarding this issue? Got the same one, cannot see my organization repositories (I can see only user)

**Comment by apazga** [comment]
Hi! Same here using Bitbucket.
I tried setting permissions for Read, Write or Admin, but repositories list is empty on Docker Hub after choosing the organization.
Any update on this?

**Comment by chrispatrick** [comment]
We have managed to identify that the list of organisations/workspaces was being populated only with those that the connected user was an admin/owner of. I have just deployed a fix to change that to list all organisations that the connected user is a member of and has admin rights on at least one repo in the organisation.

This should fix the issue, but do let us know if there are any further issues.
